src/mutator.rs

```diff
@@ -254,7 +254,7 @@ impl Scope {
     }
 
     /// Deferred destruction and deallocation of heap-allocated object `ptr`.
-    pub unsafe fn defer_drop<T: Send + 'static>(&self, ptr: Ptr<T>) {
+    pub unsafe fn defer_drop<'s, T: Send + 's>(&'s self, ptr: Ptr<'s, T>) {
```

Comment:
If we want to allow arbitrary lifetimes, we should warn the programmer about the possible pitfalls of not using `Send + 'static` data, as described [here](https://github.com/crossbeam-rs/crossbeam/pull/57#issuecomment-197055562) and generally in that thread. I'm still in favour of only allowing `Send + 'static`, but I understand that it might be too restrictive.

Reply:
I agree with you..  It seems a tricky problem...

Reply:
Yeah, I'd rather use `Send + 'static` here. Is there a particular example of this type bound being too cumbersome to deal with?

Reply:
I saw `rsdb` needs more relaxed lifetime requirement here: https://github.com/spacejam/rsdb/blob/master/src/stack.rs#L158  I think it is exactly the case that is discussed in the issue @Vtec234 referenced.